dotnet/src/webdriver/Script.cs

```diff
@@ -0,0 +1,99 @@
+// <copyright file="Script.cs" company="WebDriver Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Threading.Tasks;
+using WebDriverBiDi;
+using WebDriverBiDi.Log;
+using WebDriverBiDi.Session;
+
+namespace OpenQA.Selenium
+{
+    /// <summary>
+    /// Provides a mechanism to access WebDriverBiDi events in script and log domains.
+    /// </summary>
+    [Obsolete("This class is in beta and may change in future releases.")]
+    internal class Script : IScript
+    {
+        private readonly BiDiDriver biDiDriver;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Script"/> class
+        /// </summary>
+        /// <param name="driver">Driver in use</param>
+        [Obsolete("This class is in beta and may change in future releases.")]
+        public Script(WebDriver driver)
+        {
+            this.biDiDriver = driver.BiDiDriver;
+            this.biDiDriver.Log.EntryAdded += OnEntryAdded;
+        }
+
+        /// <summary>
+        /// Add and remove handlers for console messages.
+        /// </summary>
+        [Obsolete("This event is in beta and may change in future releases.")]
+        public event EventHandler<EntryAddedEventArgs> ConsoleMessageHandler;
```

Comment:
This design is pretty much correct, but typical .NET API design names events with verbs, usually past tense for "something just happened," or present tense with "-ing" for "something is about to happen." In this case, you'd want something like `ConsoleMessageWritten` and `JavaScriptErrorMessageWritten`. Yes, I know it's bikeshedding, and we can negotiate about the names.

Reply:
What about `ConsoleMessageHanding`?

Reply:
Both Playwright & PuppeteerCSharp use `page.Console +=` as public facing API
I'm reading this inenglish as "Add event to Page Console"
So this PR implementation would be: "Add event to Console Message Handler"
Which seems equivalent?
It's also more closely following the naming conventions we had on the board at the Dev Summit